### DISCUSSION TITLE
How to use JenaText with FusekiMain?

### DISCUSSION THREAD

**Discussion by kinow** [apache/jena#1950] [Q&A] [answered]
Hi,

I have a dataset configured for Skosmos that's similar to what's described in this page: <https://jena.apache.org/documentation/query/text-query.html#text-dataset-assembler>

But when I start FusekiMain (in Docker Compose) with:

```
command: --update --desc=/fuseki/skosmos.ttl /skosmos
```

the container fails to initialize with the stack trace:

```bash
skosmos-web     | [Mon Jul 03 17:39:21.315012 2023] [core:notice] [pid 8] AH00094: Command line: '/usr/sbin/apache2 -D FOREGROUND'
skosmos-fuseki  | org.apache.jena.sparql.ARQException: Found two matches: var ?root -> http://base/#text_dataset, http://base/#tdb_dataset_readwrite
skosmos-fuseki  |   at org.apache.jena.sparql.util.QueryExecUtils.getAtMostOne(QueryExecUtils.java:395)
skosmos-fuseki  |   at org.apache.jena.sparql.util.graph.GraphUtils.findRootByType(GraphUtils.java:213)
skosmos-fuseki  |   at org.apache.jena.sparql.core.assembler.AssemblerUtils.build(AssemblerUtils.java:138)
skosmos-fuseki  |   at org.apache.jena.sparql.core.assembler.AssemblerUtils.build(AssemblerUtils.java:132)
skosmos-fuseki  |   at arq.cmdline.ModAssembler.create(ModAssembler.java:68)
skosmos-fuseki  |   at arq.cmdline.ModDatasetAssembler.createDataset(ModDatasetAssembler.java:41)
skosmos-fuseki  |   at org.apache.jena.fuseki.main.cmds.FusekiMain.processModulesAndArgs(FusekiMain.java:370)
skosmos-fuseki  |   at org.apache.jena.cmd.CmdArgModule.process(CmdArgModule.java:39)
skosmos-fuseki  |   at org.apache.jena.cmd.CmdMain.mainMethod(CmdMain.java:86)
skosmos-fuseki  |   at org.apache.jena.cmd.CmdMain.mainRun(CmdMain.java:56)
skosmos-fuseki  |   at org.apache.jena.cmd.CmdMain.mainRun(CmdMain.java:43)
skosmos-fuseki  |   at org.apache.jena.fuseki.main.cmds.FusekiMain.run(FusekiMain.java:134)
skosmos-fuseki  |   at org.apache.jena.fuseki.main.cmds.FusekiMainCmd.main(FusekiMainCmd.java:47)
```

Reading our Java code, it looks to me like the Assembler will try to load a single `RDFDataset` (or similar type) from the given configuration, but we have the text dataset and the tdb2 dataset, thus the error.

At the same page in Jena docs, there's a comment that's similar to what I explained above, I believe:

>since the assembler contains two dataset definitions, one for the text dataset, one for the base data. Therefore, the application needs to identify the text dataset by it’s URI http://localhost/jena_example/#text_dataset.

Is there a way to initialize Fuseki Main with a assembler file for Jena Text and indicate what's the main dataset?

Thank you!

-Bruno

**Comment by afs** [comment, marked as answer]
Try using `--config`. That walks from the server node to find datasets or search for `rdf:type fuseki:Service`.

`--desc` is a shortcut - it is for publishing a single dataset at a single name with the default services (c.f. `--file` and `--mem`).
It ignores `fuseki:Service`.

It exists for other command e.g. `tdbquery`.

**Reply by kinow** [reply]
That worked! Thanks!
